rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -324,6 +324,18 @@ bool Rir2Pir::compileBC(const BC& bc, Opcode* pos, Opcode* nextPos,
                                                fs, ast)));
                 },
                 insertGenericCall);
+        } else if (monomorphic && bc.bc == Opcode::call_implicit_ &&
+                   TYPEOF(monomorphic) == BUILTINSXP) {
+            // TODO implement support for call_builtin_ with names
+            Value* expected = insert(new LdConst(monomorphic));
+            Value* given = callee;
+            if (ldfun)
+                given = insert(new LdVar(ldfun->varName, ldfun->env()));
+            Value* t = insert(new Identical(given, expected));
+            auto cp = insert.addCheckpoint(srcCode, pos, stack);
+            insert(new Assume(t, cp));
+            pop();
```

Comment:
Why there is not a framestate here?
Btw, this code is extremely similar to the lambda in compileClosure(), would be good to factorize.
In a first sight seems that the else if could be include inside the above if managing some conditions (I mean would be more clear if we separate the differences inside one big if for monomorphics).

Reply:
framestate on calls is used only for inlining. builtins cannot be inlined...